### PR TITLE
DOCSP-43089: Security landing page

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-43089
Staging - https://deploy-preview-69--docs-c.netlify.app/security/

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [ ] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/security.txt [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+~~~~~~~~~~~~~~
+
+The following code shows how to pass AWS credentials in a connection URI
+to authenticate with ``MONGODB-AWS``:
+
+.. literalinclude:: /includes/security/authentication.c
+   :language: c
+   :copyable: true
+   :start-after: // start-aws-connection-uri
+   :end-before: // end-aws-connection-uri
```
Comment (mcmorisi) [inline source/security.txt:128]: Question: Are both code samples in the `MONGODB-AWS` section meant to be identical? Wondering if something maybe got overlooked in the earlier PR that contained the code that you're pulling from for this PR.

**Review by mcmorisi — COMMENTED** [review]
One question!

File: source/security.txt (hunk 1 above)
Comment (norareidy) [reply source/security.txt:128]: Oh good catch, I don't think the code examples (other than the connection URI credentials one) should have the `<AWS IAM access key ID>:<AWS IAM secret access key>` placeholders. I'll remove them here and on the Authentication page

File: source/security.txt [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+To learn more about SCRAM-SHA-1 authentication, see :ref:`c-scram-sha-1` in
+the Authentication guide.
+
+MONGODB X.509
+-------------
+
+The following code shows how to create a connection URI to authenticate by using
+the ``X.509`` authentication mechanism:
+
+.. literalinclude:: /includes/security/authentication.c
```
Comment (kevinAlbs) [inline source/security.txt:90]: Suggest removing `:<percent-encoded password>` from the X509 example code. A username may be optionally passed, but a password is rejected.

The [auth spec](https://github.com/mongodb/specifications/blob/c84488cb9cf185e31869547a6378a701124b4665/source/auth/auth.md#mongodb-x509) notes "password MUST NOT be specified." for MONGODB-X509. Using the URI `mongodb://u:p@host/?authMechanism=MONGODB-X509` with the C driver results in this error:
```
Error parsing URI: ''MONGODB-X509' authentication mechanism does not accept a password'
```

**Review by kevinAlbs — APPROVED** [review]
LGTM with a fix to the MONGOB-X509 code example.

**Pull request merged**